docs/set-up.asciidoc

```diff
@@ -39,32 +40,278 @@ include::./custom-stack.asciidoc[]
+===== `node -r elastic-apm-node/start ...`
+
+Another way to start the agent is with the `-r elastic-apm-node/start` https://nodejs.org/api/cli.html#-r---require-module[command line option to node]. This will import and start the APM agent before your application code starts. This method allows you to enable the agent _without touching any code_. This is the recommended start method for <<lambda,monitoring AWS Lambda functions>>.
+
+[source,bash]
+----
+node -r elastic-apm-node/start app.js
+----
+
+The `-r, --require` option can also be specified via the https://nodejs.org/api/cli.html#node_optionsoptions[`NODE_OPTIONS` environment variable]:
+
+[source,bash]
+----
+# export ELASTIC_APM_...  # Configure the agent with envvars.
+export NODE_OPTIONS='-r elastic-apm-node/start'
+node app.js
+----
+
+
+[[start-option-separate-init-module]]
+===== separate APM init module
+
+If you want to avoid <<start-esm-imports,the gotcha with hoisted ES modules>> but still want the flexibility of passing a config object to the <<apm-start,agent start method>>, then a good option is to write a separate JavaScript or TypeScript module that starts the agent, and import *that* init module at the top of your main file. For example:
+
+[source,ts]
+----
+// initapm.ts
+import apm from 'elastic-apm-node';
+apm.start({
+  serverUrl: 'https://...',
+  secretToken: '...',
+  // ...
 })
 ----
-* Require the `elastic-apm-node/start` script:
-+
+
+[source,ts]
+----
+// main.ts
+import 'initapm'
+
+// Application code starts here.
+----
+
+
+[[start-gotchas]]
+==== Start Gotchas
+
+This section shows some sometimes subtle surprises starting the APM agent with some technologies. A general troubleshooting tip for using the Elastic Node.js APM agent with any build tool/system that produces compiled JavaScript is to **look at the compiled JavaScript** to see what is actually being executed by `node`.
+
+[[start-esm-imports]]
+===== Hoisted ES module imports
+
+When using a compiler/transpiler (such as https://babeljs.io/docs/en/[Babel] or https://esbuild.github.io[esbuild]) to translate from code using ES modules (`import ...` statements) to CommonJS modules (`require(...)`) there is a common surprise for starting the Elastic Node.js APM agent. Running Babel on the following produces code that does *not* work as expected:
+
+[source,js]
+----
+import apm from 'elastic-apm-node';
+apm.start() // This does not work.
+
+import http from 'http';
+// ...
+----
+
+Babel translates this to the equivalent of:
+
+[source,js]
+----
+var apm = require('elastic-apm-node');
+var http = require('http');
+apm.start() // This is started too late.
+// ...
+----
+
+All imports are "hoisted" to the top of a module, properly following ECMAScript module (ESM) semantics. However, the `apm.start()` is now too late: *after* the `http` module has been imported. This is why <<start-option-require-start-module,the `elastic-apm-node/start` module>> was added as a start option. The following will work:
```

Comment:
I think an em dash works better than a colon here.

```suggestion
All imports are "hoisted" to the top of a module, properly following ECMAScript module (ESM) semantics. However, the `apm.start()` is now too late—*after* the `http` module has been imported. This is why <<start-option-require-start-module,the `elastic-apm-node/start` module>> was added as a start option. The following will work:
```

Reply:
Sounds good to me. Interesting (well, "interesting" if you are text converter pedant, nerd, or just really bored) that asciidoc (and GitHub markdown) will convert double-dash, `--`, to an emdash. However asciidoc has a bug where if there are no spaces around it, then it will not do that conversion.